geode-core/src/main/java/org/apache/geode/distributed/internal/InternalConfigurationPersistenceService.java

```diff
@@ -841,8 +842,16 @@ private Configuration readConfiguration(File groupConfigDir) throws IOException
     return configuration;
   }
 
-  private String getDeployedBy() {
-    Subject subject = cache.getSecurityService().getSubject();
+  String getDeployedBy() {
+    Subject subject = null;
+    try {
+      subject = cache.getSecurityService().getSubject();
+    } catch (AuthenticationRequiredException e) {
```

Comment:
Can the "getDeployedBy()" be called other than "offline command like start locator"; can this be valid exception in those cases that needs to be handled. Sorry, looking at the code I could not make it if its called in offline mode.

Reply:
only for offline commands we can get this kind of exception. For online commands, there is always a subject present.